libraries/lib-wave-track/SequenceSampleCache.cpp

```diff
@@ -0,0 +1,77 @@
+#include "SequenceSampleCache.h"
+
+#include <cassert>
+
+SequenceSampleCache::SequenceSampleCache(
+   SourceSampleGetterFn fn, sampleCount sequenceLength, size_t maxBlockSize)
+    : mGetterFn(std::move(fn))
+    , mMaxBlockSize(maxBlockSize)
+    , mSequenceLength(sequenceLength)
```

Comment:
The sequence of member initializers does not agree with the sequence that the class declares the memebrs.

It's legal but will compile to execute in the sequence of the members, not the sequence of the initializers seen here.

Some compilers warn about that.

In this case it doesn't matter because there are no side effects or ordering dependencies among the initializers.  But it is good habit always to make the sequences agree.

Reply:
Missed that, thanks.